packages/smooth_app/lib/pages/product/simple_input_page.dart

```diff
@@ -211,6 +222,18 @@ class _SimpleInputPageState extends State<SimpleInputPage> {
         uniqueId: uniqueId,
       ),
     );
+
+    // We go and chek in the local database if the product is
+    // already in the database. If it is, we update the fields of the product.
+    //And if it is not, we create a new product with the fields of the changed product.
+    // and we insert it in the database. (Giving the user an immediate feedback)
+    if (cachedProduct == null) {
+      daoProduct.put(changedProduct);
+      provider.set(changedProduct);
+    } else {
+      daoProduct.put(cachedProduct);
+      provider.set(cachedProduct);
+    }
```

Comment:
The comments are not 100% relevant anymore and the code is a bit clumsy here; perhaps a better naming (or comments) of `cachedProduct` and `changedProduct` would help too (what about `fullProduct` and `onlyCurrentChangeProduct`, or `localProduct` and `minimalistProduct`)?

```dart
final Product upToDateProduct = fullProduct ?? onlyCurrentChangeProduct;
await daoProduct.put(upToDateProduct);
provider.set(upToDateProduct);
```
```suggestion

    final Product upToDateProduct = cachedProduct ?? changedProduct;
    await daoProduct.put(upToDateProduct);
    provider.set(upToDateProduct);
```